jina/executors/indexers/keyvalue.py

```diff
@@ -23,8 +23,20 @@ class _WriteHandler:
     """
 
     def __init__(self, path, mode):
-        self.body = open(path, mode)
-        self.header = open(path + '.head', mode)
+        self.path = path
+        self.mode = mode
+        self.body = open(self.path, self.mode)
+        self.header = open(self.path + '.head', self.mode)
```

Comment:
Why are we opening the files in `__init__`? Shouldn't we only open them in `__enter__`?

Reply:
To avoid the concerns from @nan-wang , we avoid opening and closing at everybatch which could damage performance. But we still check the need for opening

Reply:
I see. So we're kinda using "half" of what a ctx mger should do (open and close within in the context, right)?